File: apps/nexora-muslim/mobile/lib/infrastructure/nexora-core/identity.ts
import { env } from '@/lib/config/env';
import type { AuthSession, NexoraUser } from '@/lib/auth/types';

export type CurrentIdentity = {
  id: string;
  email: string | null;
  displayName: string;
  avatarUrl: string | null;
  phoneNumber: string | null;
  timezone: string;
  locale: string;
  status: string;
};

function getIdentityUrl() {
  return `${env.nexoraCoreUrl.replace(/\/$/, '')}/api/v1/identity/me/`;
}

export async function resolveNexoraIdentity(session: AuthSession): Promise<AuthSession> {
  if (session.user.provider !== 'firebase') return session;

  const response = await fetch(getIdentityUrl(), {
    headers: {
      Accept: 'application/json',
      Authorization: `Bearer ${session.accessToken}`,
    },
  });

  if (!response.ok) {
    throw new Error(`Nexora Core identity request failed (${response.status}).`);
  }

  const identity = (await response.json()) as CurrentIdentity;
  const user: NexoraUser = {
    id: identity.id,
    provider: 'firebase',
    providerSubject: session.user.providerSubject,
    email: identity.email ?? undefined,
    displayName: identity.displayName || undefined,
  };

  return { ...session, user };
}

File: services/nexora-api/api/v1/identity.py
from __future__ import annotations

from rest_framework.permissions import IsAuthenticated
from rest_framework.response import Response
from rest_framework.views import APIView

from apps.identity.authentication import FirebaseIdentityAuthentication
from apps.identity.models import NexoraUser


class CurrentIdentityView(APIView):
    """Return the canonical NEXORA identity resolved from the Firebase bearer token."""

    authentication_classes = [FirebaseIdentityAuthentication]
    permission_classes = [IsAuthenticated]

    def get(self, request):
        user: NexoraUser = request.user
        return Response(
            {
                "id": str(user.id),
                "email": user.email,
                "displayName": user.display_name,
                "avatarUrl": user.avatar_url or None,
                "phoneNumber": user.phone_number or None,
                "timezone": user.timezone,
                "locale": user.locale,
                "status": user.status,
            }
        )
